components/script/dom/blob.rs

```diff
@@ -270,21 +267,15 @@ impl BlobMethods for Blob {
                 match bytes {
                     Ok(b) => {
                         let cx = promise.global().get_cx();
-                        unsafe {
-                            rooted!(in(*cx) let mut array_buffer = ptr::null_mut::<JSObject>());
-                            assert!(ArrayBuffer::create(
-                                *cx,
-                                CreateWith::Slice(&b),
-                                array_buffer.handle_mut()
-                            )
-                            .is_ok());
-
-                            promise.resolve_native(&*array_buffer);
+                        let result = run_array_buffer_data_algorithm(cx, b);
+
+                        match result {
+                            Ok(FetchedData::ArrayBuffer(a)) => promise.resolve_native(&a),
+                            Err(e) => promise.reject_error(e),
+                            _ => {},
```

Comment:
nit: Here you can actually panic with an appropriate message, like "Unexpected result from run_array_buffer_data_algorithm".